StabilityPackage.wl

```diff
@@ -63,13 +82,57 @@ SelectSingleRadius[data_,ri_]:=Association[
 "mids"->data["mids"] (*Cos\[Theta] bin midpoints*)
 ]
 
+Options[siPotential]={"xflavor"-> False};
+siPotential[data_,OptionsPattern[]]:=Block[{nubardensity,nudensity,nuxdensity,\[Mu],\[Mu]b,\[Mu]x,n,tot},
+n=Length[data["mids"]];
+
+nudensity[dt_]:= Sum[Sum[data["lotsodo"][[1,f,dt,dp]]/ (h (data["freqmid"][[f]]) (Abs[data["muss"][[dt+1]]-data["muss"][[dt]]])),{f,1,Length[data["freqs"]]-1}],{dp,1,2}];
+nubardensity[dt_]:= Sum[Sum[data["lotsodo"][[2,f,dt,dp]]/ (h (data["freqmid"][[f]]) (Abs[data["muss"][[dt+1]]-data["muss"][[dt]]])),{f,1,Length[data["freqs"]]-1}],{dp,1,2}];
+
+If[OptionValue["xflavor"],
+nuxdensity[dt_]:= Sum[Sum[0.25 data["lotsodo"][[3,f,dt,dp]]/ (h (data["freqmid"][[f]]) (Abs[data["muss"][[dt+1]]-data["muss"][[dt]]])),{f,1,Length[data["freqs"]]-1}],{dp,1,2}]
+,
+nuxdensity[dt_]:=0.
+];
+
+
+\[Mu]=munits Table[nudensity[i]*(data["muss"][[i+1]]-data["muss"][[i]]),{i,1,n},{j,1,n}];
+\[Mu]b=munits Table[nubardensity[i]*(data["muss"][[i+1]]-data["muss"][[i]]),{i,1,n},{j,1,n}];
+\[Mu]x=munits Table[nuxdensity[i]*(data["muss"][[i+1]]-data["muss"][[i]]),{i,1,n},{j,1,n}];
+
+
+tot=(Tr[\[Mu]]+Tr[\[Mu]b]+2Tr[\[Mu]x]);
+Return[tot]
+]
+
+(*This function finds the asymetry factor between the electron (anti)neutrinos and the x (anti)neutrinos *)
+Bfactor[data_]:=Block[{n,nudensity,nubardensity,nuxdensity,\[Mu],\[Mu]b,\[Mu]x,B,Bb},
+n=Length[data["mids"]];
+
+nudensity[dt_]:= Sum[Sum[data["lotsodo"][[1,f,dt,dp]]/ (h (data["freqmid"][[f]]) (Abs[data["muss"][[dt+1]]-data["muss"][[dt]]])),{f,1,Length[data["freqs"]]-1}],{dp,1,2}];
+nubardensity[dt_]:= Sum[Sum[data["lotsodo"][[2,f,dt,dp]]/ (h (data["freqmid"][[f]]) (Abs[data["muss"][[dt+1]]-data["muss"][[dt]]])),{f,1,Length[data["freqs"]]-1}],{dp,1,2}];
+nuxdensity[dt_]:= Sum[Sum[0.25 data["lotsodo"][[3,f,dt,dp]]/ (h (data["freqmid"][[f]]) (Abs[data["muss"][[dt+1]]-data["muss"][[dt]]])),{f,1,Length[data["freqs"]]-1}],{dp,1,2}];
+
+\[Mu]=munits Table[nudensity[i]*(data["muss"][[i+1]]-data["muss"][[i]]),{i,1,n},{j,1,n}];
+\[Mu]b=munits Table[nubardensity[i]*(data["muss"][[i+1]]-data["muss"][[i]]),{i,1,n},{j,1,n}];
+\[Mu]x=munits Table[nuxdensity[i]*(data["muss"][[i+1]]-data["muss"][[i]]),{i,1,n},{j,1,n}];
+
+B=Tr[\[Mu]x]/Tr[\[Mu]];
+Bb=Tr[\[Mu]x]/Tr[\[Mu]b];
+
+Return[{B,Bb}];
+];
+
+
+
 (*This function creates the hamiltonians based on the data set, and returns them in a "general" way that is readable.  
 Returns 9 arguments with index,
 1,3,5,7 = H,\[Rho],A,\[Delta]H
 2,4,6,8 = Hb,\[Rho]b,Ab,\[Delta]Hb
 9=HsiRadial
 *)
-buildHamiltonians[data_,testE_,hi_]:=Module[{n,\[Theta],name11,name12,name21,name22,\[Rho],\[Rho]b,A,Ab,Hm,Hvac,\[Mu],\[Mu]b,Hsi,H,Hb,\[Delta]H,\[Delta]Hb,nudensity,nubardensity,Ve,\[Omega],HsiRad},(
+Options[buildHamiltonians]={"dh"-> True,"xflavor"-> False};
```

Comment:
What does dh do? Perhaps rename something like "do_perturb_H"?

Reply:
nevermind - see below - I think it is better to not include this functionality for simplicity